### PR TITLE
ENH: Add Dockerfile and shell scripts

### PR DESCRIPTION
**Pull request by ElDeveloper**
This is the Dockerfile that powers the Docker image for QIIME running on
BaseSpace, the shell scripts are used by the callback.js files to start
the processing of the data.

@chesterkuo, the Dockerfile included here is part of your repository found in https://github.com/chesterkuo/docker-gpgpu , are you OK with us redistributing
the file under the BSD-2 license? I couldn't find a license in your repository.

### CODE REVIEW

**Comment by chesterkuo** [comment]
Sure.

Thanks,
Chester
2015/5/28 上午5:28 於 "Yoshiki Vázquez Baeza" notifications@github.com 寫道：

**Comment by ElDeveloper** [comment]
Thanks @chesterkuo!

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM ubuntu:14.04
+#QIIME basic installation env v0.1
```
Comment (gregcaporaso) [inline docker/Dockerfile]: Could you re-word this to say *QIIME base install* (as that's the specific term that we use to define this configuration)?

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM ubuntu:14.04
+#QIIME basic installation env v0.1 
+MAINTAINER Chester Kuo <chester.kuo@gmail.com>
```
Comment (gregcaporaso) [inline docker/Dockerfile]: Should this be updated since it's in this repo?

**Comment by ElDeveloper** [comment]
@gregcaporaso, please continue making comments, I'll amend this in a following PR. Didn't realize big brother was watching :eyes: ...

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+RUN apt-get update && apt-get -y install curl
+RUN apt-get -y --no-install-recommends install cmake mercurial git make subversion
+RUN apt-get -y install gcc g++ gdb
+RUN apt-get -y install python python-dev python-pip vim
+RUN apt-get -y install pkg-config libpng-dev libfreetype6-dev
+RUN apt-get -y install libopenblas-base libopenblas-dev libatlas-base-dev gfortran libblas-dev liblapack-dev mklibs
+
+RUN mkdir -p /tmp
```
Comment (gregcaporaso) [inline docker/Dockerfile]: Maybe a different directory? For example, ``$HOME/temp``, and then set that in the qiime_config, which you could do with:

```bash
echo "temp_dir $HOME/temp" > $HOME/.qiime_config
```

Comment (ElDeveloper) [inline docker/Dockerfile]: Sounds like a reasonable suggestion, but I guess the real location where
we would want the temp folder, is wherever the EC2 instance running this
docker image has some scratch space, right?. I'll have to check the
documentation and see if I can find something about this.

Comment (chesterkuo) [inline docker/Dockerfile]: Looks great.

Go
2015/5/28 上午7:46 於 "Greg Caporaso" <notifications@github.com> 寫道：

**Pull request merged**